lib/containers/common.pm

```diff
@@ -28,35 +28,49 @@ our @EXPORT = qw(install_podman_when_needed install_docker_when_needed allow_sel
   test_container_runtime test_container_image scc_apply_docker_image_credentials scc_restore_docker_image_credentials);
 
 sub install_podman_when_needed {
+    my $host_os = shift;
+    my @pkgs    = qw(podman);
     if (script_run("which podman") != 0) {
-        if (is_sle '>=15') {
-            add_suseconnect_product('sle-module-containers');
+        if ($host_os == 'centos') {
+            assert_script_run "dnf -y install @pkgs", timeout => 160;
+        }
+        else {
+            if (is_sle '>=15') {
+                add_suseconnect_product('sle-module-containers');
+            }
+            push(@pkgs, 'podman-cni-config') if is_jeos();
+            push(@pkgs, 'apparmor-parser')   if is_leap("=15.1");    # bsc#1123387
+            zypper_call "in @pkgs";
         }
-
-        my @pkgs = qw(podman);
-        push(@pkgs, 'podman-cni-config') if is_jeos();
-        push(@pkgs, 'apparmor-parser')   if is_leap("=15.1");    # bsc#1123387
-        zypper_call "in @pkgs";
         assert_script_run('podman info');
     }
 }
 
 sub install_docker_when_needed {
-    if (is_microos) {
-        # Docker should be pre-installed in MicroOS
-        die 'Docker is not pre-installed.' if zypper_call('se -x --provides -i docker');
+    my $host_os = shift;
+
+    if ($host_os == 'centos') {
+        assert_script_run "dnf config-manager --add-repo=https://download.docker.com/linux/centos/docker-ce.repo";
+        # if podman installed allowerasing to solve conflicts
```

Comment:
```suggestion
        # if podman installed allow erasing to solve conflicts
```

Reply:
or if you wanted to put exact flag name than something like 
```suggestion
        # if podman installed use flag "--allowerasing" to solve conflicts
```

current version looks like typo , it is not obvious that you meant flag name

Reply:
Done